src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs

```diff
@@ -1725,7 +1725,7 @@ public async Task SynchronousReadsCanBeDisallowedGlobally()
                 Assert.Equal(CoreStrings.SynchronousReadsDisallowed, ioEx2.Message);
 
                 var buffer = new byte[5];
-                var length = await context.Request.Body.ReadUntilEndAsync(buffer).DefaultTimeout();
+                var length = await context.Request.Body.FillEntireBufferUntilEndAsync(buffer).DefaultTimeout();
```

Comment:
I think I messed up my last suggestion. I'll just apply it myself this time.

```suggestion
                var length = await context.Request.Body.FillBufferUntilEndAsync(buffer).DefaultTimeout();
```